src/foam/java/Class.js

```diff
@@ -242,11 +265,26 @@ foam.CLASS({
 
       this.constants.forEach(function(c) { o.out(c, '\n'); });
 
+      var extendedCls = self.SUPER_CLASSES[self.extends];
       this.fields.sort(function(o1, o2) {
-        return foam.Number.compare(o1.order, o2.order);
-      }).forEach(function(f) { if ( ! self.isEnum || ! f.static )  o.out(f, '\n'); });
+              return foam.Number.compare(o1.order, o2.order);
+            }).forEach(function(f) { if ( ! self.isEnum || ! f.static )  o.out(f, '\n'); });
 
-      this.methods.forEach(function(f) { o.out(f, '\n'); });
+      var self = this;
+      this.methods.forEach(function(f) {
+        if ( extendedCls != undefined ) {
+          var superMethod = extendedCls.methods.find(obj => {
```

Comment:
// Check that parent class already defines the same method, so we can skip outputting it here.